[KHRGA-51] Updated DT vector types (#140)
KhronosGroup/SYCL_Reference: source/iface/vector-types.rst

Vector types
************

SYCL provides a cross-platform class template that works
efficiently on SYCL devices as well as in host C++ code.
This type allows sharing of vectors between the host and its
SYCL devices. The vector supports member functions that allow
construction of a new vector from a swizzled set of component
elements.

> ``vec<typename DataT``, int NumElements> is a vector type that compiles

Reviewer: ```suggestion
``vec<typename DataT, int NumElements>`` is a vector type that compiles
```

Author: Fixed with : https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

> down to a SYCL backend built-in vector types on SYCL devices, where

Reviewer: ```suggestion
down to a SYCL backend built-in vector type on SYCL devices, where
```

> possible, and provides compatible support on the host or when it is

Reviewer: ```suggestion
possible, and provides compatible support on the host when 
```

Reviewer: I've opened [this ](https://github.com/KhronosGroup/SYCL-Docs/pull/522) around these two proposed changes

Author: As I understand agreed removal of full first sentence in the second paragraph: 
_``vec<typename DataT, int NumElements>`` is a vector type that compiles
down to a SYCL backend built-in vector type on SYCL devices, where
possible, and provides compatible support on the host or when it is
not possible._ 
Done with commit : https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

   template <typename dataT, int numElements>
   class vec;

(constructors)
==============

::

   vec();

   explicit constexpr vec(const DataT& arg);

   template <typename... ArgTN> constexpr vec(const ArgTN&... args);

   constexpr vec(const vec<DataT, NumElements>& rhs);

   vec(vector_t nativeVector);

> Default construct a vector with element type ``DataT``

Reviewer: ```suggestion
The first options default constructs a vector with element type ``DataT``
```

Author: Done with https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6

Available only when: ``NumElements > 1``.

Return an instance of the implementation-defined intermediate class
template ``__swizzled_vec__`` representing an index sequence made
up of the upper half of this SYCL ``vec`` which can be used to apply the
swizzle in a valid expression as described
in |SYCL_SPEC_SWIZZLED_VEC_CLASS|. When ``NumElements == 3``, this
SYCL ``vec`` is treated as though ``NumElements == 4``
with the fourth element undefined.

``odd``
=======

::

   __swizzled_vec__ odd() const

Available only when: ``NumElements > 1``.

Return an instance of the implementation-defined intermediate class
template ``__swizzled_vec__`` representing an index sequence made
up of the odd indexes of this SYCL ``vec`` which can be used to apply the
swizzle in a valid expression as described
in |SYCL_SPEC_SWIZZLED_VEC_CLASS|. When ``NumElements == 3``, this
SYCL ``vec`` is treated as though ``NumElements == 4``
with the fourth element undefined.

``even``
========

::

   __swizzled_vec__ even() const

Available only when: ``NumElements > 1``.

Return an instance of the implementation-defined intermediate class
template ``__swizzled_vec__`` representing an index sequence made

> up of the odd indexes of this SYCL ``vec`` which can be used to apply the

Reviewer: ```suggestion
up of the even indexes of this SYCL ``vec`` which can be used to apply the
```

Author: Fixed with https://github.com/KhronosGroup/SYCL_Reference/pull/140/commits/10a0e11a4a1cfa16ba887d881ad24f7a806151b6
